### ISSUE TITLE
DotNetCore Build misses Sources settings 

### ISSUE THREAD

**Issue by Meberem** [cake-build/cake#2165]
I suspect it is very similar to [the DotNetCorePublish issue](https://github.com/cake-build/cake/issues/2119). The latest docs on [the dotnet build command ](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build?tabs=netcore2x) suggest that the --source options should be passed through

### What You Are Seeing?
error CS0117: 'DotNetCoreBuildSettings' does not contain a definition for 'Sources'

### What is Expected?
Executing: "C:/Program Files/dotnet/dotnet.exe" build --sources "..."

### What version of Cake are you using?
0.27.2

### Are you running on a 32 or 64 bit system?
x64

### What environment are you running on?  Windows? Linux? Mac?
Windows & Centos 7.2

### Are you running on a CI Server?  If so, which one?
Bamboo

### How Did You Get This To Happen? (Steps to Reproduce)
./build.sh or ./build.ps1 with

```
DotNetCoreBuild(pathToProject, new DotNetCoreBuildSettings
{
	Sources = nugetSources,
});
```

**Comment by coder2213** [comment]
@augustoproiete 

Please provide cake command to recreate this issue.

**Comment by augustoproiete** [comment]
@coder2213 There's no special Cake command to repro this issue. As @Meberem described, [`DotNetCoreBuildSettings`](https://github.com/cake-build/cake/blob/v1.1.0/src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs) currently does not expose a property that allows specifying the equivalent [`--source`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build?tabs=netcore2x#options) argument when calling `dotnet build`.

@Meberem seems to be on the right track in that this would be similar to what has been done for [DotNetCorePublishSettings](https://github.com/cake-build/cake/blob/v1.1.0/src/Cake.Common/Tools/DotNetCore/Publish/DotNetCorePublishSettings.cs#L157-L163) resolved via [#2119](https://github.com/cake-build/cake/issues/2119).

**Comment by eoehen** [comment]
@augustoproiete I would be happy if I can solve this issue.
